From 5ff1a5a4af7b644fd537822a289977b76a18cb08 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 13 Dec 2021 07:31:13 +0200
Subject: [PATCH 1/6] s32gen1: sramc: Adjust SRAMC range to delete only the
 requested region.

This decreases the end of the calculated range to avoid situations
when more bytes are cleared than requested.

Issue: ALB-8270
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32g/s32g_sramc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/nxp/s32/s32g/s32g_sramc.c b/plat/nxp/s32/s32g/s32g_sramc.c
index 840d81ee8..65714565c 100644
--- a/plat/nxp/s32/s32g/s32g_sramc.c
+++ b/plat/nxp/s32/s32g/s32g_sramc.c
@@ -83,7 +83,7 @@ static void clear_sram_range(struct sram_ctrl *c, uintptr_t start_addr,
 	end_addr -= c->min_sram_addr;
 
 	start_offset = c->a53_to_sramc_offset(start_addr);
-	end_offset = c->a53_to_sramc_offset(end_addr);
+	end_offset = c->a53_to_sramc_offset(end_addr) - 1;
 
 	clear_sramc_range(base, start_offset, end_offset);
 }
-- 
2.17.1

